### PR TITLE
Check return value flow of metrics YAML generator

### PR DESCRIPTION
**Pull request by KVS85**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.8

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: utils/generateAllMetricsDocumentation.py [hunk 1]
```diff
@@ -123,40 +123,41 @@ def print_usage():
...
-
-        YAMLS.append(y)   # for later dump
-
-        # Check a few things in the yaml:
-        for attr in ["name", "help", "exposedBy", "description", "unit",\
-                     "type", "category", "complexity", "introducedIn"]:
-            if not attr in y:
-                print("YAML file '" + filename + \
-                      "' does not have required attribute '" + attr + "'")
+        if not bad:
```
Comment (neunhoef) [inline utils/generateAllMetricsDocumentation.py:126]: I think this is not meant. The `continue` above sorts out this file already, and we certainly do not want to skip validation of the current file if `bad` was set in a previous loop iteration...

File: utils/generateAllMetricsDocumentation.py [hunk 2]
```diff
@@ -123,40 +123,41 @@ def print_usage():
...
+            try:
+                y = load(s, Loader=Loader)
+            except YAMLError as err:
+                print("Could not parse YAML file '" + filename + "', error:\n" + str(err))
                 bad = True
+
         if not bad:
```
Comment (neunhoef) [inline utils/generateAllMetricsDocumentation.py:133]: Same here, as far as I see.

File: utils/generateAllMetricsDocumentation.py [hunk 3]
```diff
@@ -123,40 +123,41 @@ def print_usage():
...
-            if not attr in y:
-                print("YAML file '" + filename + \
-                      "' does not have required attribute '" + attr + "'")
+        if not bad:
+            try:
+                y = load(s, Loader=Loader)
+            except YAMLError as err:
+                print("Could not parse YAML file '" + filename + "', error:\n" + str(err))
                 bad = True
+
```
Comment (neunhoef) [inline utils/generateAllMetricsDocumentation.py:132]: There probably should be a `continue` here...

File: utils/generateAllMetricsDocumentation.py (hunk 1 above)
Comment (KVS85) [reply utils/generateAllMetricsDocumentation.py:125]: Forgot to delete. That was there initially.

File: utils/generateAllMetricsDocumentation.py (hunk 2 above)
Comment (neunhoef) [reply utils/generateAllMetricsDocumentation.py:132]: Nonsense, please ignore.

File: utils/generateAllMetricsDocumentation.py (hunk 3 above)
Comment (neunhoef) [reply utils/generateAllMetricsDocumentation.py:131]: Nonsense, please ignore.

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**